### PR TITLE
fix(components): [table] table-column expand nested table wrong style

### LINKED ISSUE #8521: [Component] [table] When table-column expand nested table show wrong style
Bug Type: **`Style`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.6`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36 Edg/100.0.1185.50`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbiAgaW1wb3J0IHsgcmVmLCB2ZXJzaW9uIGFzIHZ1ZVZlcnNpb24sY29tcHV0ZWQgfSBmcm9tIFwidnVlXCI7XG4gIGltcG9ydCB7IHZlcnNpb24gYXMgRXBWZXJzaW9uIH0gZnJvbSBcImVsZW1lbnQtcGx1c1wiO1xuICBpbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiO1xuXG4gIGNvbnN0IGJvcmRlcj1yZWYodHJ1ZSlcbiAgY29uc3QgdGFibGVEYXRhID0gQXJyYXkuZnJvbSh7IGxlbmd0aDogMiB9LCAoXywgaSkgPT4gKHtcbiAgICBmb286IGksXG4gICAgYmFyOiBpICsgMSxcbiAgICBiYXo6IGkgKyAyLFxuICB9KSk7XG4gIGNvbnN0IGNoaWxkVGFibGVEYXRhID0gY29tcHV0ZWQoKCk9PiBBcnJheS5mcm9tKHsgbGVuZ3RoOiA1IH0sIChfLCBpKSA9PiAoe1xuICAgIGZvbzogXCLihpAgc2VlXCIsXG4gICAgYmFyOiBg4oaQIGJvcmRlciAke2JvcmRlci52YWx1ZT8nbm90JzonJ30gY29ycmVjdCDihpJgLFxuICAgIGJhejogXCJzZWUg4oaSXCIsXG4gIH0pKSk7XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICBzd2l0Y2ggZWwtdGFibGUgYm9yZGVyIGF0dHI6XG4gIDxlbC1zd2l0Y2ggdi1tb2RlbD1cImJvcmRlclwiPjwvZWwtc3dpdGNoPlxuICA8YnIvPlxuICA8YnIvPlxuICA8ZWwtdGFibGUgOmJvcmRlcj1cImJvcmRlclwiIDpkYXRhPVwidGFibGVEYXRhXCIgZGVmYXVsdC1leHBhbmQtYWxsPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gdHlwZT1cImV4cGFuZFwiPlxuICAgICAgPGRpdiBzdHlsZT1cInBhZGRpbmc6IDAgMzBweFwiPlxuICAgICAgICA8ZWwtdGFibGUgOmRhdGE9XCJjaGlsZFRhYmxlRGF0YVwiPlxuICAgICAgICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJmb29cIiBwcm9wPVwiZm9vXCI+PC9lbC10YWJsZS1jb2x1bW4+XG4gICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cImJhclwiIHByb3A9XCJiYXJcIj48L2VsLXRhYmxlLWNvbHVtbj5cbiAgICAgICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiYmF6XCIgcHJvcD1cImJhelwiPjwvZWwtdGFibGUtY29sdW1uPlxuICAgICAgICA8L2VsLXRhYmxlPlxuICAgICAgPC9kaXY+XG4gICAgPC9lbC10YWJsZS1jb2x1bW4+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cImZvb1wiIHByb3A9XCJmb29cIj48L2VsLXRhYmxlLWNvbHVtbj5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiYmFyXCIgcHJvcD1cImJhclwiPjwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJiYXpcIiBwcm9wPVwiYmF6XCI+PC9lbC10YWJsZS1jb2x1bW4+XG4gIDwvZWwtdGFibGU+XG4gIDxwPlxuICAgIDxlbC1pY29uIGNvbG9yPVwidmFyKC0tZWwtY29sb3ItcHJpbWFyeSlcIj48RWxlbWVudFBsdXMgLz48L2VsLWljb24+XG4gICAgRWxlbWVudCBQbHVzIHt7IEVwVmVyc2lvbiB9fSArIFZ1ZSB7eyB2dWVWZXJzaW9uIH19XG4gIDwvcD5cbjwvdGVtcGxhdGU+IiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce
```html
<script setup lang="ts">
  import { ref, version as vueVersion,computed } from "vue";
  import { version as EpVersion } from "element-plus";
  import { ElementPlus } from "@element-plus/icons-vue";

  const border=ref(true)
  const tableData = Array.from({ length: 2 }, (_, i) => ({
    foo: i,
    bar: i + 1,
    baz: i + 2,
  }));
  const childTableData = computed(()=> Array.from({ length: 5 }, (_, i) => ({
    foo: "← see",
    bar: `← border ${border.value?'not':''} correct →`,
    baz: "see →",
  })));
</script>

<template>
  switch el-table border attr:
  <el-switch v-model="border"></el-switch>
  <br/>
  <br/>
  <el-table :border="border" :data="tableData" default-expand-all>
    <el-table-column type="expand">
      <div style="padding: 0 30px">
        <el-table :data="childTableData">
          <el-table-column label="foo" prop="foo"></el-table-column>
          <el-table-column label="bar" prop="bar"></el-table-column>
          <el-table-column label="baz" prop="baz"></el-table-column>
        </el-table>
      </div>
    </el-table-column>
    <el-table-column label="foo" prop="foo"></el-table-column>
    <el-table-column label="bar" prop="bar"></el-table-column>
    <el-table-column label="baz" prop="baz"></el-table-column>
  </el-table>
  <p>
    <el-icon color="var(--el-color-primary)"><ElementPlus /></el-icon>
    Element Plus {{ EpVersion }} + Vue {{ vueVersion }}
  </p>
</template>
```
### What is Expected?

When parent table has border attribute and child table doesn't has border attribute, child shouldn't have border style

### What is actually happening?

Child table has `border`

### Additional comments

未命名

### PR DESCRIPTION
**Pull request by liunnn1994**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Fixed https://github.com/element-plus/element-plus/issues/8521

Snipaste_2022-06-28_18-55-50
Snipaste_2022-06-28_18-56-05
Snipaste_2022-06-28_18-56-15
Snipaste_2022-06-28_18-56-32

### CODE REVIEW

**Comment by asdjgfr** [comment]
> 🧪 Playground Preview: https://element-plus.run/?pr=8522 Please comment the example via this playground if needed.

Couldn't save in https://element-plus.run/?pr=8522. Add example in [https://element-plus.run](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbiAgaW1wb3J0IHsgcmVmLCB2ZXJzaW9uIGFzIHZ1ZVZlcnNpb24sY29tcHV0ZWQgfSBmcm9tIFwidnVlXCI7XG4gIGltcG9ydCB7IHZlcnNpb24gYXMgRXBWZXJzaW9uLEVsVGFibGUgfSBmcm9tIFwiZWxlbWVudC1wbHVzXCI7XG4gIGltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI7XG5cbiAgY29uc3QgcGFyZW50Qm9yZGVyPXJlZih0cnVlKVxuICBjb25zdCBjaGlsZEJvcmRlcj1yZWYodHJ1ZSlcbiAgY29uc3QgdGFibGVEYXRhID0gQXJyYXkuZnJvbSh7IGxlbmd0aDogMiB9LCAoXywgaSkgPT4gKHtcbiAgICBmb286IGksXG4gICAgYmFyOiBpICsgMSxcbiAgICBiYXo6IGkgKyAyLFxuICB9KSk7XG4gIGNvbnN0IGNoaWxkVGFibGVEYXRhID0gY29tcHV0ZWQoKCk9PiBBcnJheS5mcm9tKHsgbGVuZ3RoOiA1IH0sIChfLCBpKSA9PiAoe1xuICAgIGZvbzogXCLihpAgc2VlXCIsXG4gICAgYmFyOiBg4oaQIGJvcmRlciBjb3JyZWN0IOKGkmAsXG4gICAgYmF6OiBcInNlZSDihpJcIixcbiAgfSkpKTtcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIHN3aXRjaCBwYXJlbnQgZWwtdGFibGUgYm9yZGVyIGF0dHI6XG4gIDxlbC1zd2l0Y2ggdi1tb2RlbD1cInBhcmVudEJvcmRlclwiPjwvZWwtc3dpdGNoPlxuICA8YnIvPlxuICBzd2l0Y2ggY2hpbGQgZWwtdGFibGUgYm9yZGVyIGF0dHI6XG4gIDxlbC1zd2l0Y2ggdi1tb2RlbD1cImNoaWxkQm9yZGVyXCI+PC9lbC1zd2l0Y2g+XG4gIDxici8+XG4gIDxici8+XG4gIDxlbC10YWJsZSA6Ym9yZGVyPVwicGFyZW50Qm9yZGVyXCIgOmRhdGE9XCJ0YWJsZURhdGFcIiBkZWZhdWx0LWV4cGFuZC1hbGw+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiB0eXBlPVwiZXhwYW5kXCI+XG4gICAgICA8ZGl2IHN0eWxlPVwicGFkZGluZzogMCAzMHB4XCI+XG4gICAgICAgIDxlbC10YWJsZSA6ZGF0YT1cImNoaWxkVGFibGVEYXRhXCIgOmJvcmRlcj1cImNoaWxkQm9yZGVyXCI+XG4gICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cImZvb1wiIHByb3A9XCJmb29cIj48L2VsLXRhYmxlLWNvbHVtbj5cbiAgICAgICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiYmFyXCIgcHJvcD1cImJhclwiPjwvZWwtdGFibGUtY29sdW1uPlxuICAgICAgICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJiYXpcIiBwcm9wPVwiYmF6XCI+PC9lbC10YWJsZS1jb2x1bW4+XG4gICAgICAgIDwvZWwtdGFibGU+XG4gICAgICA8L2Rpdj5cbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiZm9vXCIgcHJvcD1cImZvb1wiPjwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJiYXJcIiBwcm9wPVwiYmFyXCI+PC9lbC10YWJsZS1jb2x1bW4+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cImJhelwiIHByb3A9XCJiYXpcIj48L2VsLXRhYmxlLWNvbHVtbj5cbiAgPC9lbC10YWJsZT5cbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgRXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctODUyMi1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTg1MjItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctODUyMi1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy04NTIyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19).

**Comment by asdjgfr** [comment]
@YunYouJun @sxzz Shall we merge this PR? This error cause something wrong in my project, THX.

File: packages/theme-chalk/src/table.scss [hunk 1]
```diff
@@ -634,4 +634,33 @@
       margin-right: 8px;
     }
   }
+
+  @include b(table) {
+    &.#{$namespace}-table--border {
+      .#{$namespace}-table__cell {
+        border-right: getCssVar('table-border') !important;
```
Comment (YunYouJun) [inline packages/theme-chalk/src/table.scss:641]: Can we remove  `!mportant`?

We should not leave styles in the component library that cannot be overridden by the user, but conflicts can be avoided with more complex selectors.

Comment (asdjgfr) [reply packages/theme-chalk/src/table.scss]: Done :D

**Comment by asdjgfr** [comment]
@YunYouJun Already removed `!mportant`, shall we merge? THX.

**Comment by YunYouJun** [comment]
Good job. 

One last tip, can you add your example to the documentation?
Or modify an existing example to make it easier for users to preview.

**Comment by asdjgfr** [comment]
> Good job.
> 
> One last tip, can you add your example to the documentation? Or modify an existing example to make it easier for users to preview.
> 
> > Just like what you wriitten in playground

@YunYouJun I modified [`Expandable row`](https://element-plus.org/en-US/component/table.html#expandable-row) part. Add switch button and update data from [Tom_and_Jerry](https://en.wikipedia.org/wiki/Tom_and_Jerry) 😋.

**Review by YunYouJun — APPROVED** [review]

**Pull request merged**